### ISSUE TITLE
"Discard modified line" does not complete without switching focus on app

### ISSUE THREAD

**Issue by Mikifarber1** [shiftkey/desktop#690]
### Search existing issues

- [X] I have searched the issue tracker and confirmed that I was unable to find existing issues for the problem I am experiencing.

### The problem

upon pressing the discard modify line for a certain line. the popup menu isn't reacting and I had to close the app and reopen it.
the operation occurred - meaning that the line was discarded.. but I had to close and open the app 

+extra feature request: only when I'm pressing the line number I can perform the "discard modify line" and honestly I didn't know that you had this option.. can you change it that when I'm pressing the mouse right-click anywhere on the line then in the popup menu (copy/expand the whole file) you will add the discard button?

### Release version

2.9.9

### Operating system

linux

**Comment by shiftkey** [comment]
@Mikifarber1 thanks for the report.

If you look at the logs you should see some messages like this:

```
2022-04-04T13:49:08.079Z - info: [ui] Executing discardChangesFromSelection: git apply --unidiff-zero --whitespace=nowarn - (took 17.471s)
```

Note that `took` value at the end there. On my machine it took over 17s, which isn't great for what's not a complex repoistory. Can you confirm you're also seeing a high number when you do this?

**Comment by shiftkey** [comment]
> +extra feature request: only when I'm pressing the line number I can perform the "discard modify line" and honestly I didn't know that you had this option.. can you change it that when I'm pressing the mouse right-click anywhere on the line then in the popup menu (copy/expand the whole file) you will add the discard button?

Please submit this to the upstream project as feedback: https://github.com/desktop/desktop

**Comment by Mikifarber1** [comment]
> @Mikifarber1 thanks for the report.
> 
> If you look at the logs you should see some messages like this:
> 
> ```
> 2022-04-04T13:49:08.079Z - info: [ui] Executing discardChangesFromSelection: git apply --unidiff-zero --whitespace=nowarn - (took 17.471s)
> ```
> 
> Note that `took` value at the end there. On my machine it took over 17s, which isn't great for what's not a complex repoistory. Can you confirm you're also seeing a high number when you do this?

when I waited long enough I was able to press another button in the pop up window and to continue to work..

**Comment by Mikifarber1** [comment]
> > +extra feature request: only when I'm pressing the line number I can perform the "discard modify line" and honestly I didn't know that you had this option.. can you change it that when I'm pressing the mouse right-click anywhere on the line then in the popup menu (copy/expand the whole file) you will add the discard button?
> 
> Please submit this to the upstream project as feedback: https://github.com/desktop/desktop

Done, https://github.com/desktop/desktop/issues/14279

**Comment by shiftkey** [comment]
Based on the symptoms I've published a new release [`2.9.12-linux4`](https://github.com/shiftkey/desktop/releases/tag/release-2.9.12-linux4) which has a different version of the Git package that I've been carrying around since the app was running on Electron 11 (and Node v12) that I think are now regressed because we're on Electron 17 (and Node v16) and no longer necessary. 

So I've undone that change and I see this operation complete immediately when running locally, without needing to interact with the app again. Let me know if that works for your situation also.

**Comment by shiftkey** [comment]
Closing this out as I believe this has been resolved by the latest release
